serverlessrepo/parser.py

```diff
@@ -122,3 +124,23 @@ def parse_application_id(text):
     """
     result = re.search(APPLICATION_ID_PATTERN, text)
     return result.group(0) if result else None
+
+
+def strip_app_metadata(template_dict):
+    """
+    Strip the "AWS::ServerlessRepo::Application" metadata section from template.
+
+    :param template_dict: SAM template as a dictionary
+    :type template_dict: dict
+    :return: stripped template content
+    :rtype: str
+    """
+    template_dict_copy = copy.deepcopy(template_dict)
+
+    # strip the whole metadata section if SERVERLESS_REPO_APPLICATION is the only key in it
+    if not [k for k in template_dict_copy.get(METADATA) if k != SERVERLESS_REPO_APPLICATION]:
+        template_dict_copy.pop(METADATA, None)
+    else:
+        template_dict_copy.get(METADATA).pop(SERVERLESS_REPO_APPLICATION, None)
+
+    return yaml_dump(template_dict_copy)
```

Comment:
If it takes a template as a dictionary, it seems weird to not return a dictionary. The caller can yaml dump it if they want.